src/yunohost/app.py

```diff
@@ -517,6 +517,10 @@ def app_upgrade(app=[], url=None, file=None, force=False):
     from yunohost.regenconf import manually_modified_files
 
     apps = app
+    # Check if disk space available
+    size = os.statvfs('/')
+    if (size.f_bavail * size.f_frsize) / 1024 <= 512000:
```

Comment:
Sooo ideally it could be nice to use https://github.com/YunoHost/yunohost/blob/dev/src/yunohost/utils/filesystem.py#L24 

(to be checked if this need to be imported at the top of the file)

```suggestion
    if free_space_in_directory("/") <= 512 * 1000 * 1000:
```

Reply:
OK, thanks for the suggestion.
Why multiply 2 times by 1000 and not 1 time by 2000?

Reply:
It's mainly cosmetics to try to improve readability ... maybe not so much in this case, but in the diagnosis code we had a similar situation so i defined `MB = 1000 * 1000` so I could write `512 * MB` ... but anyway nevermind, the important stuff here is about using the existing helper function